File: 프로그래머스/lv2/43165. 타겟 넘버/타겟 넘버.py
from collections import deque

def solution(numbers, target):
    deq = deque()
    deq2 = deque()
    deq.append(numbers[0])
    deq.append(-numbers[0])
    for i in range(1,len(numbers)):
        while deq:
            a = deq.popleft()
            deq2.append(a+numbers[i])
            deq2.append(a-numbers[i])
        while deq2:
            deq.append(deq2.popleft())
    
    return deq.count(target)
            

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.swift
import Foundation

func solution(_ numbers:[Int], _ target:Int) -> Int {
    var result = 0
    let totalCnt = numbers.count
    func recursion(_ idx: Int, _ sum: Int) {
        if idx == totalCnt {
            if sum == target {
                result += 1
            }
            return
        }
        recursion(idx+1, sum + numbers[idx])
        recursion(idx+1, sum - numbers[idx])
    }
    recursion(0, 0)
    return result
}